src/rez/pip.py

```diff
@@ -218,63 +70,52 @@ def run_pip_command(command_args, pip_version=None, python_version=None):
 
 
 def find_pip(pip_version=None, python_version=None):
-    """Find a pip exe using the given python version.
+    """Find pip.
+
+    Will revert to native pip installed with rez, if a pip rez package cannot
+    be found. In this case, None is returned.
+
+    Args:
+        pip_version (str or `Version`): Version of pip to use, or latest if None.
+        python_version (str or `Version`): Python version to use, or latest if
+            None.
 
     Returns:
         2-tuple:
-            str: pip executable;
-            `ResolvedContext`: Context containing pip, or None if we fell back
-                to system pip.
+        - str: Python executable.
+        - `ResolvedContext`: Context containing pip, or None if we fell back
+          to system pip.
     """
-    pip_exe = "pip"
+    py_exe = "python"
+    context = None
 
+    # find pip, fall back to system if rez pip package not found
     try:
         context = create_context(pip_version, python_version)
+        py_exe = context.which("python")
     except BuildError:
-        # fall back on system pip. Not ideal but at least it's something
-        from rez.backport.shutilwhich import which
-
-        pip_exe = which("pip")
-
-        if pip_exe:
-            print_warning(
-                "pip rez package could not be found; system 'pip' command (%s) "
-                "will be used instead." % pip_exe)
-            context = None
-        else:
-            raise
+        # fall back on system pip
```

Comment:
Before falling back to system (so virtualenv's python), I would try to get the latest python version (if python was not provided) available as a rez package and assume that there might be pip installed into it. If this fails too, then fallback to rez's virtualenv pip.

Would that make sense?

Reply:
I believe my latest comment addresses this - as in yes we would change things to be expecting python package to provided pip, and would be checking for this before falling back to rezified pip, then to system pip (and rezified pip just for backwards compatibility - current users could very well have python without pip, and rezified pip).